lib/engine/game/g_1824/game.rb

```diff
@@ -728,10 +728,11 @@ def float_minor!(minor, price)
         end
 
         # Modifed 1837 version as it will log incorrect received amount after
-        # formation + float.
+        # formation + float. Also use special float amount factor to handle unreserve of
+        # IPO shares, which caused a bug previously.
         def float_corporation(corporation)
           @log << "#{corporation.name} floats"
-          capitilization = corporation.par_price.price * corporation.total_ipo_shares
+          capitilization = corporation.par_price.price * corporation.float_amount_factor
```

Comment:
As the reserved shares are marked as not buyable, could you do something like this?
```suggestion
          unreserved_shares = corporation.shares.reject(:buyable).sum(&:percent) / 10
          capitilization = corporation.par_price.price * unreserved_shares
```
You then wouldn't need to track the `float_amount_factor`.

Reply:
Do you need to make the share buyable at that point?

Reply:
It need to be buyable.

In the first case the reserved share becomes a normal IPO share and can be bought from IPO during SR.
In the second case the reserved share becomes a normal share owned by a player and can be sold during SR.

There is a tricky detail in 1824 concerning SD1, UG1, KK1, the 20% in SD/UG/KK should be not be buyable. But Staatsbahn is formed, as soon as a player owns 20% or more, they will get the president's share.

In the current implementation this is handled by setting buyable to true, but then some special cases to handle so that it cannot be bought, and presidency handling. I have not looked into if the current implementation for handling this case can be simplified.